entrypoint.sh

```diff
@@ -39,30 +39,44 @@ unset_prev_secrets() {
 
 # Install op-cli
 install_op_cli() {
+  # Create a temporary directory where the CLI is installed
   OP_INSTALL_DIR="$(mktemp -d)"
   if [[ ! -d "$OP_INSTALL_DIR" ]]; then
     echo "Install dir $OP_INSTALL_DIR not found"
     exit 1
   fi
   export OP_INSTALL_DIR
   echo "::debug::OP_INSTALL_DIR: ${OP_INSTALL_DIR}"
+
+  # Get the latest stable version of the CLI
+  OP_CLI_VERSION="v$(curl https://app-updates.agilebits.com/check/1/0/CLI2/en/2.0.0/N -s | jq -r .version)"
+
   if [[ "$OSTYPE" == "linux-gnu"* ]]; then
-    ARCHITECTURE=""
-    if [[ "$(uname -m)" == "x86_64" ]]; then
-      ARCHITECTURE="amd64"
-    elif [[ "$(uname -m)" == "aarch64" ]]; then
-      ARCHITECTURE="arm64"
-    else
-      echo "Unsupported architecture"
+    # Get runner's architecture
+    ARCH=$(uname -m)
+    if [[ "$(getconf LONG_BIT)" = 32 ]]; then
+      ARCH="386"
+    elif [[ "$ARCH" == "x86_64" ]]; then
+      ARCH="amd64"
+    elif [[ "$ARCH" == "aarch64" ]]; then
+      ARCH="arm64"
+    fi
+
+    if [[ "$ARCH" != "386" ]] && [[ "$ARCH" == "amd64" ]] && [[ "$ARCH" == "arm" ]] && [[ "$ARCH" == "arm64" ]]; then
+      echo "Unsupported architecture for the 1Password CLI: $ARCH."
       exit 1
     fi
```

Comment:
Can we simplify this check by getting rid of this `if` chain and making the `echo` error message/`exit 1` the last condition in the above `if` block?

Reply:
There is a mistake in the conditional. Nice catch!
Fixed it with [f195e2d](https://github.com/1Password/load-secrets-action/pull/49/commits/f195e2df03423d2e879db28f05869acaf88aa6b3).